### ISSUE TITLE
Not displaying anything

### ISSUE THREAD

**Issue by Astrus1234** [AtomLinter/linter-flake8#240]
Hello,
I have installed linter-flake8 with atom. Atom automatically installed linter. My problem is that he doesnt detect any error with python script. The strange things if I run flake8 from command line it detect multiple error and specify in which file errors occured.

Here is some screenshoot : http://image.noelshack.com/fichiers/2016/34/1472320581-capture-d-ecran-2016-08-27-a-19-55-57.png
http://image.noelshack.com/fichiers/2016/34/1472331581-capture-d-ecran-2016-08-27-a-22-58-42.png http://image.noelshack.com/fichiers/2016/34/1472331581-capture-d-ecran-2016-08-27-a-22-58-59.png http://image.noelshack.com/fichiers/2016/34/1472331582-capture-d-ecran-2016-08-27-a-22-59-05.png http://image.noelshack.com/fichiers/2016/34/1472331582-capture-d-ecran-2016-08-27-a-22-59-20.png
PS: I'm running on macosx

**Comment by Astrus1234** [comment]
Okay now I have specified the exact path of flake8 in the setting of atom and its working on new file. But on old file that I have already written it's not working at all...
Here is an exemple : 
http://image.noelshack.com/fichiers/2016/34/1472398327-capture-d-ecran-2016-08-28-a-17-31-40.png

You can see that there is no error displayed but when I execute flake8 on this particular file it gave me this output : 
http://pastebin.com/t8PRNdz1

**Comment by Arcanemagus** [comment]
Just looking at it, I would guess that it's an issue with the character set of the file, there are some hacky things in place that _should_ be fixing this, but might not be.

Are you able to share the full file it isn't working on?

_PS: Sorry about the late response, been flooded with notifications on here and still working through everything._

**Comment by Astrus1234** [comment]
Hey,

Here is the file you requested :
[fonctions.py.zip](https://github.com/AtomLinter/linter-flake8/files/452566/fonctions.py.zip)

No problem for late response, I was loosing hope that I could run flake8 on atome :P

**Comment by Astrus1234** [comment]
Hey,

Here is the file you requested :
[fonctions.py.zip](https://github.com/AtomLinter/linter-flake8/files/452575/fonctions.py.zip)

No problem for late response, I was loosing hope that I could run flake8 on atome :P

**Comment by Arcanemagus** [comment]
Oh, also, can you post the output of `flake8 --version`?

**Comment by Astrus1234** [comment]
flake8 --version
3.0.4 (pycodestyle: 2.0.0, mccabe: 0.5.2, pyflakes: 1.2.3, flake8-docstrings: 1.0.2, pydocstyle: 1.0.0) CPython 3.5.2 on Darwin

**Comment by Arcanemagus** [comment]
I can't reproduce unfortunately... I'm getting the same output in the CLI as within Atom.

If you open Atom's Developer Tools and put a breakpoint [here](https://github.com/AtomLinter/linter-flake8/blob/v1.13.2/lib/main.coffee#L228), what does `result` contain?

**Comment by Astrus1234** [comment]
I should open it with ctrl+alt+i on atom right ?
But after I'm kinda confused, I dont know where to look to find main.coffee ?

**Comment by kyan001** [comment]
```
File "c:\kyan_software\python3\lib\site-packages\flake8\processor.py", line 248, in line_for
    adjusted_line_number = line_number - 1
TypeError: unsupported operand type(s) for -: 'str' and 'int'"
```

my flake8 version is 
> 3.0.4 (pycodestyle: 2.0.0, pyflakes: 1.2.3, mccabe: 0.5.2) CPython 3.5.2 on Windows

**Comment by Arcanemagus** [comment]
@kyan001 Do you see the same thing when running `type foo.py | flake8`?

**Comment by vasa-chi** [comment]
I too don't see any errors from flake8. When debugging, as was suggested in [this comment](https://github.com/AtomLinter/linter-flake8/issues/240#issuecomment-244491471), I got this error stack:

```
"Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.4/bin/flake8", line 11, in <module>
    sys.exit(main())
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/main/cli.py", line 16, in main
    app.run(argv)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/main/application.py", line 316, in run
    self._run(argv)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/main/application.py", line 300, in _run
    self.run_checks()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/main/application.py", line 237, in run_checks
    self.file_checker_manager.start(files)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/checker.py", line 364, in start
    self.make_checkers(paths)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/checker.py", line 275, in make_checkers
    for argument in paths
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/checker.py", line 278, in <listcomp>
    if should_create_file_checker(filename)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/checker.py", line 404, in __init__
    self.processor = self._make_processor()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/checker.py", line 414, in _make_processor
    return processor.FileProcessor(self.filename, self.options)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/processor.py", line 59, in __init__
    self.lines = self.read_lines()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/processor.py", line 271, in read_lines
    lines = self.read_lines_from_stdin()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/processor.py", line 307, in read_lines_from_stdin
    return utils.stdin_get_value().splitlines(True)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/flake8/utils.py", line 73, in stdin_get_value
    stdin_value = sys.stdin.read()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd0 in position 422: ordinal not in range(128)"
```

When calling flake8 from console, everything works fine.
flake8 version: `3.0.4 (pyflakes: 1.2.3, pycodestyle: 2.0.0, mccabe: 0.5.2) CPython 3.4.4 on Darwin`

**Comment by ztolley** [comment]
Same here. Using latest atom 1.10.2 , linter flake 1.13.4 and I get no errors shown
When I set a breakpoint I get 
"Fatal Python error: Py_Initialize: unable to load the file system codec ImportError: No module named 'encodings' Current thread 0x00007fffb35e43c0 (most recent call first):"

Is it using the system python and not the virtual env python? I started atom from the command line after pointing to my virtual env (which is ./env)

**Comment by kyan001** [comment]
@Arcanemagus I've found the cause. When using String.split() or String.splitlines(), Some Chinese UTF-8 characters are wrong encoded and pass to the next step, so there is a `surrogates not allowed` exception raised.

I  met this problem in one of my 3 computers. Besides, when using flake8 console command, this problem is not shown.

My solution is to add encode() with ignore unknown chars in build_ast in flake8/processor.py line 185
```python
def build_ast(self):
    """Build an abstract syntax tree from the list of lines."""
    return compile(''.join(self.lines).encode('utf8','ignore'), '', 'exec', PyCF_ONLY_AST)
```

**Comment by Astrus1234** [comment]
@kyan001
How do you modify the file processor.py ? I dont find the file

**Comment by kyan001** [comment]
@Astrus1234 I installed flake8 by pip, so it's in my pythonfolder/Lib/site-packages/flake8/processor.py
(win10 + Python 3.5.2)

I located this file because it's shown in atom develop tools (Ctrl + Alt + i), goto source tab, find top/file://../AppData/Local/.atom/packages/linter-flake8/lib/main.coffee, and make a breakpoint on line 235
```coffee
if (result.stderr and result.stderr.length and atom.inDevMode())
```
where you can see the errors in result.stderr, and debug by print() which is in result.stdout.

**Comment by kyan001** [comment]
@vasa-chi  I just upgrade my flake8 from 2.6.2 to 3.0.4, and get same problem as yours.
My solution is go back to flake8 2.6.2
```sh
pip uninstall flake8
pip install flake8==2.6.2
```

**Comment by ztolley** [comment]
I get the same error when I switch to flake 2.6.2
"Fatal Python error: Py_Initialize: unable to load the file system codec ImportError: No module named 'encodings' Current thread 0x00007fffb558d3c0 (most recent call first):"

**Comment by keisuke-nakata** [comment]
@kyan001 's [solution](https://github.com/AtomLinter/linter-flake8/issues/240#issuecomment-247923893) works perfectly for me! Thank you very much.

**Comment by Arcanemagus** [comment]
@kyan001 can you please file that as an issue over on the [flake8 repository](https://gitlab.com/pycqa/flake8/issues)? That way it can be fixed for everyone 😉. Thanks for tracking this down!

**Comment by Wolfonye** [comment]
Hello everyone, i have the same issue as Astrus had initially but couldn't solve it until now. With Ubuntu i was able to get everything up and running but with Win10 nothing happens at all. I already tried switching the flake version and specifying the exact path via atom settings. I don't even get any error messages which indicate that flake couldn't be found or anything. Exist experiences for this case?

**Comment by vasa-chi** [comment]
@Wolfonye does flake8 work from command line in win10?

**Comment by Wolfonye** [comment]
@vasa-chi No it didn't it produced the following output:
`E:\Repositories\PythonFristSteps>flake8 test.py
Traceback (most recent call last):
  File "e:\programmierung\python\lib\site-packages\flake8\checker.py", line 449, in run_check
    arguments,
  File "e:\programmierung\python\lib\site-packages\flake8\processor.py", line 214, in keyword_arguments_for
    arguments[param] = getattr(self, param)
AttributeError: 'FileProcessor' object has no attribute 'physical_line'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "e:\programmierung\python\lib\runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "e:\programmierung\python\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "E:\Programmierung\Python\Scripts\flake8.exe\__main__.py", line 9, in <module>
  File "e:\programmierung\python\lib\site-packages\flake8\main\cli.py", line 16, in main
    app.run(argv)
  File "e:\programmierung\python\lib\site-packages\flake8\main\application.py", line 316, in run
    self._run(argv)
  File "e:\programmierung\python\lib\site-packages\flake8\main\application.py", line 300, in _run
    self.run_checks()
  File "e:\programmierung\python\lib\site-packages\flake8\main\application.py", line 238, in run_checks
    self.file_checker_manager.run()
  File "e:\programmierung\python\lib\site-packages\flake8\checker.py", line 343, in run
    self.run_serial()
  File "e:\programmierung\python\lib\site-packages\flake8\checker.py", line 327, in run_serial
    checker.run_checks(self.results_queue, self.statistics_queue)
  File "e:\programmierung\python\lib\site-packages\flake8\checker.py", line 568, in run_checks
    self.process_tokens()
  File "e:\programmierung\python\lib\site-packages\flake8\checker.py", line 552, in process_tokens
    self.handle_newline(token_type)
  File "e:\programmierung\python\lib\site-packages\flake8\checker.py", line 595, in handle_newline
    self.run_logical_checks()
  File "e:\programmierung\python\lib\site-packages\flake8\checker.py", line 503, in run_logical_checks
    results = self.run_check(plugin, logical_line=logical_line) or ()
  File "e:\programmierung\python\lib\site-packages\flake8\checker.py", line 455, in run_check
    exception=ae,
flake8.exceptions.PluginRequestedUnknownParameters: "hacking.core" requested unknown parameters causing 'FileProcessor' object has no attribute 'physical_line'`

But i must admit that didn't really help me to get to the core of the problem. I'm still new to this. (Command line in Ubuntu worked well, sorry for the late reply btw.)

**Comment by vasa-chi** [comment]
Well, it seems that flake8 doesn't work on your windows machine at all. You
should probably open an issue in flake8 repo.
сб, 24 сент. 2016 г. в 0:19, Wolfonye <notifications@github.com>:

**Comment by Wolfonye** [comment]
check, i will do that, thanks for helping.

**Comment by Arcanemagus** [comment]
Marking this as closed, the issue isn't in this project.
